### ISSUE TITLE
__END__ \n Data Only

### ISSUE THREAD

**Issue by mjago** [asterite/rufo#89]
A data only file:
`__END__\nData Only` formats to `\n__END__\nData Only` formats to `\n\n__END__\nData Only` at which point there are no further changes. 

Possibly related to an empty file formatting to `\n`, then no further changes (ruby doesn't complain about a zero byte file).

**Comment by asterite** [comment]
Thanks for the bug report! 😸 

About files ending with newline: I always followed that rule but didn't know why (I saw it mentioned in several places without an explanation). Today I decided to search and I found some interesting results: https://robots.thoughtbot.com/no-newline-at-end-of-file and https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline . So I think it's better to stick that newline for better compatibility with tools like `cat` and `wc`.

I wonder though if it should be configurable...
